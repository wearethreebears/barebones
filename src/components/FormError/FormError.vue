<template>
  <span
    v-if="props.error"
    :data-component="$options.name"
    :id="errorId"
    :class="part['component']"
    data-part="component"
    >{{ props.error }}</span
  >
</template>

<script lang="ts">
export default { name: "BonesFormError" };
</script>

<script setup lang="ts">
import useComponent from "@barebones/composables/useComponent";
import useError from "@barebones/composables/useError";
import { name, error } from "@barebones/components/__shared__/props/form";

import { classProps } from "@barebones-local/FormError/FormError.props";
import FormErrorLocal from "@barebones-local/FormError/FormError.parts";

const { parts } = FormErrorLocal();

const props = defineProps({
  ...classProps,
  name,
  error,
});

const { part } = useComponent(parts, props, classProps);
const { errorId } = useError(props.name);
</script>
