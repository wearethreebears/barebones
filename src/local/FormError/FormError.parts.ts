import type { Parts } from "@barebones/types/component";
import type { FormError } from "@barebones/types/components/FormError";

export default function FormErrorLocal(): FormError {
  const parts: Parts = {
    component: {},
  };

  return {
    parts,
  };
}
