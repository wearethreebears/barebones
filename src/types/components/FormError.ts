import type { Parts } from "@barebones/types/component";

export interface FormError {
  parts: Parts;
}
