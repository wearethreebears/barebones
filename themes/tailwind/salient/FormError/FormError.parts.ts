import type { Parts } from "@barebones/types/component";
import type { FormError } from "@barebones/types/components/FormError";

// Base styles

export default function FormErrorLocal(): FormError {
  const parts: Parts = {
    component: {},
  };

  return {
    parts,
  };
}
