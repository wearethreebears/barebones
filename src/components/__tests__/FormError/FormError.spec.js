import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { allComponentPartClassesToDisplayCorrectly } from "@barebones/components/__tests__/helpers";
import FormError from "@barebones/components/FormError/FormError.vue";

import FormErrorLocal from "@barebones-local/FormError/FormError.parts";
const { parts } = FormErrorLocal();


const error = "Hello world";

allComponentPartClassesToDisplayCorrectly(FormError, parts);

describe("FormError", () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = mount(FormError, {
      props: {
        error
      }
    });
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the correct slot", () => {
    expect(wrapper.html()).contains(error)
  });

});
