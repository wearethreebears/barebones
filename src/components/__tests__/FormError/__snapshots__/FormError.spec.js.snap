// Vitest Snapshot v1

exports[`FormError > renders properly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <span
        class=""
        data-component="BonesFormError"
        data-part="component"
        id="error-undefined"
      >
        Hello world
      </span>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "error": "Hello world",
            "name": undefined,
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "error": {
              "0": false,
              "1": true,
              "default": undefined,
              "required": false,
              "type": [Function],
            },
            "errorId": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": "error-undefined",
              "_value": "error-undefined",
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    [Circular],
                    Set {
                      [Circular],
                    },
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
            },
            "name": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
            "part": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": {},
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    Set {
                      [Circular],
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
            "parts": {
              "component": {},
            },
            "props": {
              "error": "Hello world",
              "name": undefined,
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "error": "Hello world",
            "name": undefined,
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "error": {
                "0": false,
                "1": true,
                "default": undefined,
                "required": false,
                "type": [Function],
              },
              "name": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            [
              "error",
            ],
          ],
          "provides": {},
          "proxy": {
            "error": {
              "0": false,
              "1": true,
              "default": undefined,
              "required": false,
              "type": [Function],
            },
            "name": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": {},
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        Set {
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "error": {
              "0": false,
              "1": true,
              "default": undefined,
              "required": false,
              "type": [Function],
            },
            "errorId": "error-undefined",
            "name": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
            "part": {},
            "parts": {
              "component": {},
            },
            "props": {
              "error": "Hello world",
              "name": undefined,
            },
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": "Hello world",
            "component": null,
            "dirs": null,
            "dynamicChildren": [],
            "dynamicProps": [
              "data-component",
              "id",
            ],
            "el": <span
              class=""
              data-component="BonesFormError"
              data-part="component"
              id="error-undefined"
            >
              Hello world
            </span>,
            "key": 0,
            "patchFlag": 11,
            "props": {
              "class": "",
              "data-component": "BonesFormError",
              "data-part": "component",
              "id": "error-undefined",
              "key": 0,
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 9,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "span",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/FormError/FormError.vue",
            "__hmrId": "854f6f4e",
            "components": {},
            "name": "BonesFormError",
            "props": {
              "error": {
                "0": false,
                "1": true,
                "default": undefined,
                "required": false,
                "type": [Function],
              },
              "name": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <span
          class=""
          data-component="BonesFormError"
          data-part="component"
          id="error-undefined"
        >
          Hello world
        </span>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "error": "Hello world",
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/FormError/FormError.vue",
          "__hmrId": "854f6f4e",
          "components": {},
          "name": "BonesFormError",
          "props": {
            "error": {
              "0": false,
              "1": true,
              "default": undefined,
              "required": false,
              "type": [Function],
            },
            "name": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <span
          class=""
          data-component="BonesFormError"
          data-part="component"
          id="error-undefined"
        >
          Hello world
        </span>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": {
    "error": {
      "0": false,
      "1": true,
      "default": undefined,
      "required": false,
      "type": [Function],
    },
    "name": {
      "0": false,
      "1": true,
      "required": true,
      "type": [Function],
    },
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <span
    class=""
    data-component="BonesFormError"
    data-part="component"
    id="error-undefined"
  >
    Hello world
  </span>,
}
`;
